src/account_x/x_account_controller.ts

```diff
@@ -2142,6 +2250,89 @@ export class XAccountController {
         };
     }
 
+    importXArchiveMedia(tweet: XArchiveTweet, archivePath: string) {
+        // Loop over all media items
+        tweet.extended_entities?.media.forEach((media: XAPILegacyTweetMedia) => {
+            const existingMedia = exec(this.db, 'SELECT * FROM tweet_media WHERE mediaID = ?', [media.id_str], "get") as XTweetMediaRow;
+            if (existingMedia) {
+                log.debug(`XAccountController.importXArchiveMedia: media already exists: ${media.id_str}`);
+                return;
+            }
+            const filename = this.saveXArchiveMedia(tweet.id_str, media, archivePath);
+            if (filename) {
+                // Index media information in tweet_media table
+                exec(this.db, 'INSERT INTO tweet_media (mediaID, mediaType, url, filename, startIndex, endIndex, tweetID) VALUES (?, ?, ?, ?, ?, ?, ?)', [
+                    media.id_str,
+                    media.type,
+                    media.url,
+                    filename,
+                    media.indices?.[0],
+                    media.indices?.[1],
+                    tweet.id_str,
+                ]);
+            }
+        });
+    }
+
+    saveXArchiveMedia(tweetID: string, media: XAPILegacyTweetMedia, archivePath: string): string | null {
+        if (!this.account) {
+            throw new Error("Account not found");
+        }
+
+        log.info(`XAccountController.saveXArchiveMedia: saving media: ${JSON.stringify(media)}`);
+
+        const mediaURL = getMediaURL(media);
+        const mediaExtension = mediaURL.substring(mediaURL.lastIndexOf(".") + 1);
+        const filename = `${media["id_str"]}.${mediaExtension}`;
+        const archiveMediaFilename = path.join(
+            archivePath,
+            "data",
+            "tweets_media",
+            `${tweetID}-${mediaURL.substring(mediaURL.lastIndexOf("/") + 1)}`
+        );
+
+        // If file doesn't exist in archive, don't save information in db
+        if (!fs.existsSync(archiveMediaFilename)) {
+            log.info(`XAccountController.saveXArchiveMedia: media file not found: ${archiveMediaFilename}`);
```

Comment:
with the videos for loading from the export, I'm hitting this log line here

Reply:
this is the log line:

`22:27:32.890 › XAccountController.saveXArchiveMedia: media file not found: /Users/redshiftzero/Documents/Cyd Dev/X/$TEST_ACCOUNT_USERNAME/tmp/data/tweets_media/1891594792125010175-kvvBMojc1QwudY9d.mp4`

the file in question is named `data/tweets_media/1891594792125010175-jq7vI73puwizA_En.mp4` in the archive

Reply:
In the `media` object, there are various bitrate options, and it seems like the one by default in the archive is the highest bitrate one

Reply:
added 7deb52c6e942b5aef168bd2f249e03fad744fbf6 which fixes for me